### ISSUE TITLE
TrainResult wrong parameters on docs

### ISSUE THREAD

**Issue by Vichoko** [PyTorchLightning/pytorch-lightning#3091]
While checking the code against the docs I noticed that the docs mention 3 positional parameters on [```pl.TrainResult```](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.core.step_result.html#pytorch_lightning.core.step_result.TrainResult) class, but the [code signature has 4](https://github.com/PyTorchLightning/pytorch-lightning/blob/bcdb750976ca2a62925fdf5201720d6b383db37e/pytorch_lightning/core/step_result.py#L472-L626). 
Also in the getting started tutorial instances this class with the first positional parameter without explaining what it uses for. This last inconsistency is key because I can't figure if [the tutorial](https://pytorch-lightning.readthedocs.io/en/latest/new-project.html#wrap-loss-in-a-trainresult-evalresult) intends to track the training loss for ```early_stop_on``` param (based on docs) or ```minimize``` param (based on source code).

PD: Also there is a typo in the tutorial part I referenced last (https://pytorch-lightning.readthedocs.io/en/latest/new-project.html#wrap-loss-in-a-trainresult-evalresult)

**Comment by Borda** [comment]
it seems very similar to #3093, mind do the same as https://github.com/PyTorchLightning/pytorch-lightning/issues/3093#issuecomment-679880672 🦝
